### DISCUSSION TITLE
self._outliers

### DISCUSSION THREAD

**Discussion by maticar92** [MaartenGr/BERTopic#1840] [General]
Hi, I hope this is not a too obvious question. When I check bertopic_model._outliers of a trained model it only outputs: 1. I checked the place where this variable gets created and its on line 267 of the _bertopic.py file as follows: 

`266    self._outliers = 1`

What is this supposed to mean. It would be very helpful if it would track which topics were assigned as outliers. But it seems like it's just a tracker to signify that outliers exist and be used to modify the index of assignments to topics accounting for outliers to being topic -1. 

If it would be helpful to track which documents are outliers I would love to give it a go with a PR, probably tracking somewhere else like in a self._outlier documents. 

The reason it would be helpful is that when saving models, or when merging models we could keep track of what was already deemed an outlier by the HDBSSCAN clustering.

**Comment by MaartenGr** [comment]
The purpose of the `self._outliers` variable is nothing more than tracking whether outliers exist in the model. If so, it is used across a number of functions to properly check and select the right topics. There are many functionalities that do not involve outliers, so it is important that we can track and potentially ignore them. 

>  If it would be helpful to track which documents are outliers I would love to give it a go with a PR, probably tracking somewhere else like in a self._outlier documents.

I'm not sure whether I understand correctly. All documents that are outliers are marked as `-1` in the resulting topic list that you can access as the output of `.fit_transform` or with `topic_model.topics_`. Couldn't you use those?

**Comment by maticar92** [comment]
I could absolutely, but sometimes those change when using the .transform() function because the documents get assigned topics using cosine similarity. 

The issue is not very salient anymore it was to keep outliers after merging models since the submodes are not saved. But you are right the ._topics_ keeps the original -1. 

Thank you.
